addons/hw_drivers/main.py

```diff
@@ -96,7 +96,10 @@ def run(self):
             _logger.warning("An error happened when trying to get the HTTPS certificate: %s",
                             certificate_details)
 
-        iot_client = WebsocketClient(helpers.get_odoo_server_url())
+        if helpers.get_odoo_server_url():
+            iot_client = WebsocketClient(helpers.get_odoo_server_url())
+        else:
+            iot_client = None
```

Comment:
```suggestion
        iot_client = helpers.get_odoo_server_url() and WebsocketClient(helpers.get_odoo_server_url())
```
would be cleaner IMHO (but I'm not sure it is equivalent and if iot_client can be `False`)